### DISCUSSION TITLE
Renaming measures

### DISCUSSION THREAD

**Discussion by InsightfulAnalytics** [TabularEditor/TabularEditor3#766] [🙋♀️TE3 Q&A] [answered]
when changing the name of a measure in tabular editor 3 and saving the changes back to power bi, the visual containing my gets an error as it is looking for the measure name before I changed the name.  

That is, when changing measure names in tabular editor 3 I need to rebuild my visuals that contain the measure that was just renamed because i get the error "the field can't be used since it is invalid".

If changing the measure name within the PBI dax editor, I don't experience this issue.

**Comment by otykier** [comment, marked as answer]
Tabular Editor cannot access the visual layer of a .pbix file, and as such, it can’t detect if measures or other objects are being used in visuals, nor can it update the visuals when an object is renamed. But even if it could, changing object names might still break other reports that point to the same dataset, Analyze in Excel workbooks, etc. As a model developer, it is your responsibility to finalize object names before publishing the dataset and sharing it with others, to avoid these types of changes later on. If you must rename an object on a dataset that is already in use, consider hiding the original object, then create a (visible) copy that uses the new name.

There is also an (unsupported) workaround using a open-source product called [pbi-tools](https://data-goblins.com/power-bi/replace-report-measures?format=amp).

**Reply by InsightfulAnalytics** [reply]
Thank you @otykier appreciate the answer.
